File: solutions/Golang/longest_substring_without_repeating_characters_brute_force.go
package main

import "fmt"

func longestSubstringWithoutRepeatingCharactersBruteForce(s string) int {
	maxLength := 0

	for i := 0; i < len(s); i++ {
		charSet := make(map[rune]bool)
		for j, c := range s[i:] {
			if charSet[c] {
				break
			}
			charSet[c] = true
			if i+j+1 > maxLength {
				maxLength = i + j + 1
			}
		}
	}

	return maxLength
}

func main() {
	s := "abcabcbb"
	result := longestSubstringWithoutRepeatingCharactersBruteForce(s)
	fmt.Println(result)
}

File: solutions/C++/longest_substring_without_repeating_characters_brute_force.cpp
#include <iostream>
#include <unordered_set>
#include <string>
#include <algorithm>

int longest_substring_without_repeating_characters_brute_force(std::string s) {
    int max_length = 0;

    for (int i = 0; i < (int)s.length(); i++) {
        std::unordered_set<char> char_set;
        for (int j = i; j < (int)s.length(); j++) {
            if (char_set.count(s[j])) {
                break;
            }
            char_set.insert(s[j]);
            max_length = std::max(max_length, j - i + 1);
        }
    }

    return max_length;
}

int main() {
    std::string s = "abcabcbb";
    std::cout << longest_substring_without_repeating_characters_brute_force(s) << std::endl;
    return 0;
}
